src/gamestate.lua

```diff
@@ -747,6 +769,131 @@ function read_player_vars(_player_obj)
+    _reset_3_addr = _player_obj.charge_3_reset_addr
+    _valid_3 = false
+  elseif _player_obj.char_str == "oro" then
+    _charge_1_addr = _player_obj.charge_3_addr
+    _reset_1_addr = _player_obj.charge_3_reset_addr
+    _player_obj.charge_1.name= "Sun Disk"
+    _valid_1 = true
+    _charge_2_addr = _player_obj.charge_5_addr
+    _reset_2_addr = _player_obj.charge_5_reset_addr
+    _player_obj.charge_2.name= "Yanma"
+    _valid_2 = true
+    _charge_3_addr = _player_obj.charge_3_addr
+    _reset_3_addr = _player_obj.charge_3_reset_addr
+    _valid_3 = false
+  elseif _player_obj.char_str == "urien" then
+    _charge_1_addr = _player_obj.charge_5_addr
+    _reset_1_addr = _player_obj.charge_5_reset_addr
+    _player_obj.charge_1.name= "Tackle"
+    _valid_1 = true
+    _charge_2_addr = _player_obj.charge_2_addr
+    _reset_2_addr = _player_obj.charge_2_reset_addr
+    _player_obj.charge_2.name= "Kneedrop"
+    _valid_2 = true
+    _charge_3_addr = _player_obj.charge_4_addr
+    _reset_3_addr = _player_obj.charge_4_reset_addr
+    _player_obj.charge_3.name= "Headbutt"
+    _valid_3 = True
+  elseif _player_obj.char_str == "remy" then
+    _charge_1_addr = _player_obj.charge_4_addr
+    _reset_1_addr = _player_obj.charge_4_reset_addr
+    _player_obj.charge_1.name= "LoV High"
+    _valid_1 = true
+    _charge_2_addr = _player_obj.charge_3_addr
+    _reset_2_addr = _player_obj.charge_3_reset_addr
+    _player_obj.charge_2.name= "Lov Low"
+    _valid_2 = true
+    _charge_3_addr = _player_obj.charge_5_addr
+    _reset_3_addr = _player_obj.charge_5_reset_addr
+    _player_obj.charge_3.name= "Rising"
+    _valid_3 = true
+  elseif _player_obj.char_str == "q" then
+    _charge_1_addr = _player_obj.charge_5_addr
+    _reset_1_addr = _player_obj.charge_5_reset_addr
+    _player_obj.charge_1.name= "Dash Atk"
+    _valid_1 = true
+    _charge_2_addr = _player_obj.charge_4_addr
+    _reset_2_addr = _player_obj.charge_4_reset_addr
+    _player_obj.charge_2.name= "Dash Low"
+    _valid_2 = true
+    _charge_3_addr = _player_obj.charge_1_addr
+    _reset_3_addr = _player_obj.charge_1_reset_addr
+    _valid_3 = false
+  elseif _player_obj.char_str == "chunli" then
+    _charge_1_addr = _player_obj.charge_5_addr
+    _reset_1_addr = _player_obj.charge_5_reset_addr
+    _player_obj.charge_1.name= "Bird Kick"
+    _valid_1 = true
+    _charge_2_addr = _player_obj.charge_1_addr
+    _reset_2_addr = _player_obj.charge_1_reset_addr
+    _valid_2 = false
+    _charge_3_addr = _player_obj.charge_1_addr
+    _reset_3_addr = _player_obj.charge_1_reset_addr
+    _valid_3 = false
+  else
+    _charge_1_addr = _player_obj.charge_1_addr
+    _reset_1_addr = _player_obj.charge_1_reset_addr
+    _valid_1 = false
+    _charge_2_addr = _player_obj.charge_1_addr
+    _reset_2_addr = _player_obj.charge_1_reset_addr
+    _valid_2 = false
+    _charge_3_addr = _player_obj.charge_1_addr
+    _reset_3_addr = _player_obj.charge_1_reset_addr
+    _valid_3 = false
+  end
```

Comment:
I have the feeling that all this code would be simpler if all the data was in a map and the code was simply accessing the right data through char_str. What do you think ?

Reply:
Also I just tested with Q and I got a third bar named rising (it comes from remy's part I guess), but i don't really get how this could happen

Reply:
Okay, I tested with all the characters and all the characters that do not have 3 types of charge are automatically completed with Remy's types

Reply:
This should be resolved now as I needed to properly unset the meter from displaying if you change characters. 

Using a data map is certainly easier to organize and I added some logic to handle the map properly.